Adyen/Components/BaseFormComponent/FormElements.swift

```diff
@@ -0,0 +1,119 @@
+//
+// Copyright (c) 2021 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import Foundation
+
+/// :nodoc:
+internal protocol BaseFormElement {
+
+    /// :nodoc:
+    associatedtype StyleType: ViewStyle
+
+    /// :nodoc:
+    associatedtype ItemType: FormItem
+
+    /// :nodoc:
+    var identifier: String { get }
+
+    /// :nodoc:
+    var style: StyleType { get }
+
+    /// :nodoc:
+    func build(_ itemBuilder: BaseFormBuilder) -> ItemType
+}
+
+/// :nodoc:
+internal struct FirstNameElement: BaseFormElement {
```

Comment:
Do we really need specifically `FirstNameElement` and `LastNameElement ` ?
What if we next have 'MiddleName', `AddressStreet`, `AddressCity` e.t.c ?

How about jut TextElement with params, so we can extend list easily?


`

Reply:
the idea, is that `FormTextInputItem` is very generic, but the `FirstNameElement` and `LastNameElement` are specific to have the corresponding item  configuration specifically using the correct localization, if I make it generic then how is it different than `FormTextInputItem `.

the idea is that `Elements` are a less abstract layer on top of `FormItems`